ansible/roles/test/files/ptftests/vrf_test.py

```diff
@@ -0,0 +1,519 @@
+
+        total_entry_hit_cnt = 0
+
+        for member in ecmp_entry:
+            total_entry_hit_cnt += port_hit_cnt.get(member, 0)
+
+        (p, r) = check_within_expected_range(test, total_entry_hit_cnt, float(total_hit_cnt)/len(dest_port_list))
+
+        logging.info("%-10s \t %-10s \t %10d \t %10d \t %10s"
+                        % ("ECMP", str(ecmp_entry), total_hit_cnt/len(dest_port_list), total_entry_hit_cnt, str(round(p, 4)*100) + '%'))
+
+        result &= r
+
+        if len(ecmp_entry) == 1 or total_entry_hit_cnt == 0:
+            continue
+
+        for member in ecmp_entry:
+            (p, r) = check_within_expected_range(test, port_hit_cnt.get(member, 0), float(total_entry_hit_cnt)/len(ecmp_entry))
+            logging.info("%-10s \t %-10s \t %10d \t %10d \t %10s"
+                            % ("LAG", str(member), total_entry_hit_cnt/len(ecmp_entry), port_hit_cnt.get(member, 0), str(round(p, 4)*100) + '%'))
+            result &= r
+
+    assert result
+
+
+def check_traffic(test, src_port, dst_ip_addr, dst_port_list, balance_port_list, ipv4=True):
+    if ipv4:
+        (pkt, masked_exp_pkt) = generate_ipv4_packet(test, dst_ip_addr)
+    else:
+        (pkt, masked_exp_pkt) = generate_ipv6_packet(test, dst_ip_addr)
+
+    send_packet(test, src_port, pkt)
+    logging.info("Sending packet from port " + str(src_port) + " to " + dst_ip_addr + ', packet_action is: ' + test.pkt_action)
+
+    if test.pkt_action == 'fwd':
+
+        logging.info("expect receive packets in " + str(dst_port_list))
+
+        (matched_index, received) = verify_packet_any_port(test, masked_exp_pkt, dst_port_list)
+        assert received
+
+        matched_port = dst_port_list[matched_index]
+        logging.info("Received packet at " + str(matched_port))
+
+        # Test traffic balancing across ECMP/LAG members
+        if len(dst_port_list) > 1 and test.balance and random.random() < test.balancing_test_ratio :
+
+            logging.info("Check IP range balancing...")
+
+            hit_count_map = {}
+
+            for i in range(0, test.balancing_test_times):
+                if ipv4:
+                    (pkt, masked_exp_pkt) = generate_ipv4_packet(test, dst_ip_addr)
+                else:
+                    (pkt, masked_exp_pkt) = generate_ipv6_packet(test, dst_ip_addr)
+
+                send_packet(test, src_port, pkt)
+                logging.info("Sending packet from port " + str(src_port) + " to " + dst_ip_addr)
+
+                (matched_index, received) = verify_packet_any_port(test, masked_exp_pkt, dst_port_list)
+                matched_port = dst_port_list[matched_index]
+                hit_count_map[matched_port] = hit_count_map.get(matched_port, 0) + 1
+
+            check_balancing(test, balance_port_list, hit_count_map)
+    else:
+        logging.info("expect not receive packet in " + str(dst_port_list))
+        verify_no_packet_any(test, masked_exp_pkt, dst_port_list)
+
+
+class FibTest(BaseTest):
```

Comment:
Is it possible to reuse existing ptftest(https://github.com/Azure/sonic-mgmt/tree/master/ansible/roles/test/files/ptftests)? Seems some functions already implemented there, idea is to avoid redundant code.

Reply:
Improved `fib_test.FibTest` and `vrf_test.FwdTest`